crates/fuel-core/src/service/adapters/executor.rs

```diff
@@ -65,3 +81,43 @@ impl fuel_core_executor::ports::RelayerPort for RelayerIterableKeyValueView {
         }
     }
 }
+
+impl NewTxWaiterPort for NewTxWaiter {
+    async fn wait_for_new_transactions(&mut self) -> WaitNewTransactionsResult {
+        tokio::select! {
+            _ = tokio::time::sleep_until(self.timeout) => {
+                WaitNewTransactionsResult::Timeout
+            }
+            res = self.receiver.changed() => {
+                match res {
+                    Ok(_) => {
+                        WaitNewTransactionsResult::NewTransaction
+                    }
+                    Err(_) => {
+                        WaitNewTransactionsResult::Timeout
+                    }
+                }
+            }
+        }
+    }
+}
+
+impl PreconfirmationSenderPort for PreconfirmationSender {
+    async fn send(
+        &self,
+        preconfirmations: Vec<PreconfirmationStatus>,
+    ) -> anyhow::Result<()> {
+        self.sender.send(preconfirmations).await.map_err(Into::into)
+    }
+
+    fn try_send(
+        &self,
+        preconfirmations: Vec<PreconfirmationStatus>,
+    ) -> Vec<PreconfirmationStatus> {
+        match self.sender.try_send(preconfirmations) {
+            Ok(()) => vec![],
+            Err(TrySendError::Closed(_)) => vec![],
+            Err(TrySendError::Full(preconfirmations)) => preconfirmations,
```

Comment:
Do we want to notify someone that the channel is closed instead of acting like all of them were sent?

Reply:
They were not send, it is why we return them back and it is up to the caller decide what he wants to do with unsent preconfirmations

Reply:
I know they weren't sent. There is just no difference in the success path and the `Closed` path to the caller.

Reply:
I changed the closed to return the preconfirmations also. In any cases closed shouldn't happens I also added a warn print

Reply:
@MitchTurner Removed behavior but added comments to explain why we are ignoring errors.